api/types/role.go

```diff
@@ -952,6 +976,27 @@ func (r *RoleV6) SetGroupLabels(rct RoleConditionType, labels Labels) {
 	}
 }
 
+// CheckAndSetDefaults checks validity of all fields and sets defaults
+func (c *SPIFFERoleCondition) CheckAndSetDefaults() error {
+	if c.Path == "" {
+		return trace.BadParameter("path: should be non-empty")
+	}
+	isRegex := strings.HasPrefix(c.Path, "^") && strings.HasSuffix(c.Path, "$")
+	if !(strings.HasPrefix(c.Path, "/") || isRegex) {
+		return trace.BadParameter(
+			"path: should start with / or be a regex expression starting with ^ and ending with $",
+		)
+	}
+	for i, str := range c.IPSANs {
+		if _, _, err := net.ParseCIDR(str); err != nil {
+			return trace.BadParameter(
+				"validating ip_sans[%d]: %s", i, err.Error(),
+			)
+		}
```

Comment:
Should you try to compile the DNSSAN expressions to prevent people from inserting bad regexes and the failures become silently

Reply:
I feel that pain and unfortunately, utils can't be imported without the go mod replaces everywhere.
Lets add a todo to refactor the regex methods out of utils and make them available elsewhere where we can import to `api`